shank-macro-impl/src/macros/program_id.rs

```diff
@@ -40,17 +40,14 @@ impl TryFrom<&[ItemMacro]> for ProgramId {
             Err(ParseError::new_spanned(
                 &matches[0].0,
                 format!(
-                    "Found more than one program id candidate: {:?}",
-                    matches
-                        .iter()
-                        .map(|x| x.1.clone())
-                        .collect::<Vec<String>>()
+                    "Found more than one program id candidate: {:?}. Specify one with -p",
```

Comment:
This makes no sense here as if shank found multiple `program_id!`s in the code then providing another one via a flag surely won't fix that problem :)

Reply:
I think it makes sense if the default behavior is to always override the program id.

If users want to be notified if they are going to accidentally override the program id, they should pass "--warn" or something

Reply:
OK, but then we'd need to indicate that clearer in the message:

```
Found more than one program id candidate: {:?}. You should either have exactly one `declare_id!` in your code or override the program id via -p.
```

Reply:
Added this exact error message + updated tests